### ISSUE TITLE
Regarding Optical Flow

### ISSUE THREAD

**Issue by chandrasekaraditya** [zkawfanx/Learning-Temporal-Consistency-for-Low-Light-Video-Enhancement-from-Single-Images#20]
Hello @zkawfanx!

In ```train.py```, you use the function ```WarpingLayerBWFlow```. Just wanted to know if the flow predicted using CMP is the backward flow, because ```warp.py``` also contains a function named ```WarpingLayerFWFlow```, which seems longer and is able to handle masks as well.

Thanks!

**Comment by zkawfanx** [comment]
Hi, thanks for you interest in our work!

Regarding the warping function used for the adjacent frame, the predicted flow using CMP is actually not limited for one warping direction. 

Since the predicted flow is randomly generated and non-existent before, it can be used as either forward flow or backward flow, theoretically. I remembered that I used the warping function that fills in the holes caused by warping, which should be the adopted backward flow warping function.

I didn't test the forward function for the whole pipeline. It should also work, I guess.
